node/cn/handler.go

```diff
@@ -1783,3 +1868,75 @@ func (pm *ProtocolManager) SetAcceptTxs() {
 func (pm *ProtocolManager) NodeType() common.ConnType {
 	return pm.nodetype
 }
+
+type sidecarReq struct {
+	peer    string
+	try     int
+	time    time.Time
+	request blobSidecarsRequestData
+}
+
+type sidecarReqManager struct {
+	list     map[common.Hash]*sidecarReq
+	cooldown time.Duration
+	maxTry   int
+	mu       sync.RWMutex
+}
+
+func (m *sidecarReqManager) add(txHash common.Hash, request blobSidecarsRequestData) {
+	m.mu.Lock()
+	defer m.mu.Unlock()
+	m.list[txHash] = &sidecarReq{peer: "", try: 0, time: time.Now().Add(-m.cooldown), request: request}
+}
+
+func (m *sidecarReqManager) get(txHash common.Hash) *sidecarReq {
+	m.mu.RLock()
+	defer m.mu.RUnlock()
+	return m.list[txHash]
+}
+
+func (m *sidecarReqManager) update(txHash common.Hash, peer string) {
+	m.mu.Lock()
+	defer m.mu.Unlock()
+	// no longer keep the request if the try count is too high
+	if m.list[txHash].try+1 >= m.maxTry {
```

Comment:
```suggestion
	if m.list[txHash].try > m.maxTry {
```

Reply:
`m.list[txHash].try` starts with zero. `m.list[txHash].try+1` is exactly try count in this time. So if it is max, we no longer keep the request.